qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AddUserAndAddRolesToUserSteps.java

```diff
@@ -0,0 +1,159 @@
+        goToLoginPage().login(NURSE_USERNAME, NURSE_PASSWORD);
+    }
+
+    private void reLoginAsAdmin() throws InterruptedException {
+        goToLoginPage().loginAsAdmin();
+    }
+    
+    private void fillInRoleModules(Map<String, Integer> roleModules) {
+        roleModules.put("roleStrings.Organizational:SystemAdministrator", 2);
+        roleModules.put("roleStrings.SystemDeveloper", 9);
+        roleModules.put("roleStrings.Application:AdministersSystem", 1);
+        roleModules.put("roleStrings.Organizational:Doctor", 3);
+        roleModules.put("roleStrings.Organizational:Nurse", 4);
+        roleModules.put("roleStrings.Organizational:RegistrationClerk", 3);
+    }
+	
+    @Before(RunTest.HOOK.SELENIUM_USER)
+    public void visitHomePage() {
+        initiateWithLogin();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_USER)
+    public void destroy() throws InterruptedException {
+        quit();
+    }
+	
+    @Given("a user clicks on system administration app")
+    public void userClicksOnSystemAdministrationApp() {
+        manageUserPage = homePage.goToAdministration().clickOnManageUsers();
+    }
+	
+    @Then("the system loads manage users page")
+    public void systemLoadsManageUsersPage() {
+        assertTrue(textExists("User Management"));
+    }
+	
+    @When("a user fills in person details")
+    public void userFillsInPersonDetails() {
+        manageUserPage.clickOnAddUser().createNewPerson().fillInPersonName("Super", "Nurse", NURSE_USERNAME, NURSE_PASSWORD);
+    }
+	
+    @When("a user clicks the add user link")
+    public void userClicksOnTheAddUserLink() {
+        addEditUserPage = manageUserPage.clickOnAddUser();
+        addEditUserPage.createNewPerson();
+    }
+	
+    @And("a user assigns roles to the created user")
+    public void userAssignsRolesToUser() throws InterruptedException {
+        Map<String, Integer> roleModules = new HashMap<>();
+        fillInRoleModules(roleModules);
+        String oldRole = null;
+        for (Entry<String, Integer> role : roleModules.entrySet()) {
+             manageUserPage.assignRolesToUser(oldRole, role.getKey(), NURSE_USERNAME);
+
+        reLoginAsUser();
+        if (homePage.numberOfAppsPresent() != role.getValue()) {
+            throw new AssertionError("role " + role + " doesn't have matching number of accessible applications: "
+                		+ "should be: " + role.getValue() + "is: " + homePage.numberOfAppsPresent());
+        }
+
+        reLoginAsAdmin();
+        oldRole = role.getKey();
+        homePage.goToAdministration().clickOnManageUsers();
+        }
+    }
+	
+    @Then("the system loads the create new user page")
+    public void systemLoadsCreateNewUserPage() {
+        assertTrue(textExists("Add/Edit User"));
+    }
+    
+    @And("a user checks whether the form can save with empty fields")
+    public void checkIfFormCanSaveWithEmptyFields() {
+      addEditUserPage.saveUser();
```

Comment:
```suggestion
       addEditUserPage.saveUser();
```
We should consider all spaces and Tabs the same.

Reply:
ooh okay...